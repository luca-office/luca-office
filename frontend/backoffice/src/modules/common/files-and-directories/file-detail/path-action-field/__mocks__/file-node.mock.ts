import {FileNode} from "shared/components"
import {BinaryType, NodeType} from "shared/enums"

export const fileNodeMock: FileNode = {
  binaryFileId: "spfosdkfsd",
  binaryFileUrl: "url@url.de",
  binaryType: BinaryType.Image,
  id: "sfpoksdf-fposjfd",
  name: "test-file",
  parentId: "sfopksdf-fsdpfosdk",
  tags: [],
  type: NodeType.File
}
