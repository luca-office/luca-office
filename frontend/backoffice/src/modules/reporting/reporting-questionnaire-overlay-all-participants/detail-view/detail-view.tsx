import {CSSInterpolation} from "@emotion/serialize"
import {sumBy} from "lodash-es"
import * as React from "react"
import {
  Card,
  CardContent,
  CardFooter,
  CardHeader,
  DetailViewFooter,
  Icon,
  ReportingFinalScoreQuestionnaireDetailView,
  Text
} from "shared/components"
import {IconName} from "shared/enums"
import {QuestionnaireSurveyResultsForParticipantQuery_questionnaireSurveyResultsForParticipant_questionResults as QuestionResult} from "shared/graphql/generated/QuestionnaireSurveyResultsForParticipantQuery"
import {
  FreetextQuestionCodingCriterion,
  Questionnaire,
  QuestionnaireSurveyResultsForParticipant,
  QuestionResultsByQuestionId,
  SurveyInvitationLight
} from "shared/models"
import {CustomStyle, spacingMedium, TextSize} from "shared/styles"
import {LucaTFunction} from "shared/translations"
import {find} from "shared/utils"
import {QuestionDetailView} from "./question-detail-view"

export interface Props extends CustomStyle {
  readonly t: LucaTFunction
  readonly questionnaire: Questionnaire
  readonly questionnaireSurveyResultsForParticipants: QuestionnaireSurveyResultsForParticipant[]
  readonly questionResultsByQuestion: QuestionResultsByQuestionId
  readonly freetextQuestionCodingCriteriaForQuestionnaire: FreetextQuestionCodingCriterion[]
  readonly selectedNodeId: UUID
  readonly updateSelectedNodeId: (id: UUID) => void
  readonly surveyInvitations: SurveyInvitationLight[]
}

export const DetailView: React.FC<Props> = props => {
  const {
    t,
    customStyles,
    questionnaire,
    questionnaireSurveyResultsForParticipants,
    freetextQuestionCodingCriteriaForQuestionnaire,
    questionResultsByQuestion,
    selectedNodeId,
    updateSelectedNodeId,
    surveyInvitations
  } = props

  const selectedQuestion = find(question => question.id === selectedNodeId, questionnaire.questions)

  return (
    <Card hasShadow css={customStyles}>
      <CardHeader hasGreyBackground hasShadow>
        <Icon hasSpacing name={IconName.Student} />
        <Text size={TextSize.Medium}>
          {t("reporting_scoring__scenario_details_header", {count: questionnaireSurveyResultsForParticipants.length})}
        </Text>
      </CardHeader>
      <CardContent customStyles={styles.cardContent}>
        {selectedQuestion
          .map(question => (
            <QuestionDetailView
              t={t}
              question={question}
              questionResult={questionResultsByQuestion[question.id]}
              surveyInvitations={surveyInvitations}
              questionnaireSurveyResultsForParticipants={questionnaireSurveyResultsForParticipants}
              freetextQuestionCodingCriteria={freetextQuestionCodingCriteriaForQuestionnaire.filter(
                criterion => criterion.questionId === question.id
              )}
            />
          ))
          .getOrElse(
            <ReportingFinalScoreQuestionnaireDetailView
              t={t}
              questionnaire={questionnaire}
              questionResultsByQuestion={questionResultsByQuestion}
              navigateToQuestion={updateSelectedNodeId}
              showAverageScore={true}
            />
          )}
      </CardContent>
      <CardFooter customStyles={styles.cardFooter}>
        {selectedQuestion
          .map(question => questionFooter(t, questionResultsByQuestion[question.id]))
          .getOrElse(questionnaireFooter(t, questionResultsByQuestion))}
      </CardFooter>
    </Card>
  )
}

const questionnaireFooter = (t: LucaTFunction, questionResultsByQuestion: QuestionResultsByQuestionId) => {
  const questionResults = Object.values(questionResultsByQuestion)
  const score = sumBy(questionResults, result => result.score)
  const maxScore = sumBy(questionResults, result => result.maximumScore)
  const averageScore = sumBy(questionResults, result => result.averageScore)

  return (
    <DetailViewFooter
      showAverageScoreAsProgress
      customLabelKey="rating__average_score_label"
      t={t}
      averageScore={averageScore}
      score={score}
      maxScore={maxScore}
    />
  )
}

const questionFooter = (t: LucaTFunction, questionResult: QuestionResult) => (
  <DetailViewFooter
    showAverageScoreAsProgress
    t={t}
    averageScore={questionResult.averageScore}
    customLabelKey="rating__average_score_label"
    score={questionResult.score}
    maxScore={questionResult.maximumScore}
  />
)

const styles: Record<string, CSSInterpolation> = {
  cardContent: {
    flex: "1 1 0",
    padding: spacingMedium,
    boxSizing: "border-box",
    overflow: "auto"
  },
  cardFooter: {
    display: "flex",
    flexDirection: "column",
    alignItems: "stretch"
  }
}
