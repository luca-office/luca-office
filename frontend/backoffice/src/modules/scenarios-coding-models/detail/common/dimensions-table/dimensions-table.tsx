import {css} from "@emotion/react"
import * as React from "react"
import {
  Button,
  Column,
  ColumnProps,
  Columns,
  deleteEntityButtonStyles,
  DeleteOrArchiveEntityButton,
  Heading,
  Label,
  TableContainer,
  Text,
  Tooltip
} from "shared/components"
import {ButtonVariant, HeadingLevel, IconName} from "shared/enums"
import {ScoringType} from "shared/graphql/generated/globalTypes"
import {DeleteEntityHook, NavigationConfig} from "shared/models"
import {
  CustomStyle,
  Flex,
  fontColorLight,
  FontWeight,
  spacingHuger,
  spacingSmall,
  textEllipsis,
  TextSize
} from "shared/styles"
import {LucaI18nLangKey, useLucaTranslation} from "shared/translations"
import {isEmpty, toPercent} from "shared/utils"
import {Route} from "../../../../../routes"
import {useDimensionsTable} from "./hooks/use-dimensions-table"

interface Props extends CustomStyle {
  readonly entities: DimensionTableEntity[]
  readonly labelKey: LucaI18nLangKey
  readonly maximumScore: number
  readonly placeholder?: JSX.Element
  readonly disabled?: boolean
  readonly isItemsTable?: boolean
  readonly onAddClick: () => void
  readonly onSortClick: () => void
  readonly onEntityClick: (id: UUID) => void
  readonly deleteEntityConfig: {
    readonly deleteEntityHook: DeleteEntityHook
    readonly navigateTo: NavigationConfig<Route>
  }
  readonly isReadOnly: boolean
}

export interface DimensionTableEntity {
  readonly id: string
  readonly title: string
  readonly itemsCount?: number
  readonly maxScore: number
  readonly scoringType?: ScoringType
  readonly position?: number
}

export const DimensionsTable: React.FC<Props> = ({
  customStyles,
  entities,
  labelKey,
  maximumScore,
  placeholder,
  disabled,
  isItemsTable,
  onEntityClick,
  onAddClick,
  isReadOnly,
  deleteEntityConfig,
  onSortClick
}) => {
  const {t} = useLucaTranslation()

  const {navigate} = useDimensionsTable()

  const scoringTypeColumn: ColumnProps<DimensionTableEntity> = {
    header: (
      <div css={Flex.row}>
        <Heading level={HeadingLevel.h3} fontWeight={FontWeight.Bold}>
          {t("coding_models__detail_items_scoring_type")}
        </Heading>
      </div>
    ),
    key: "scoringType",
    content: entity => (
      <Column>
        {entity.scoringType === ScoringType.Analytical
          ? t("coding_models__detail_items_scoring_type_analytic")
          : t("coding_models__detail_items_scoring_type_holistic")}
      </Column>
    )
  }

  const columns: ColumnProps<DimensionTableEntity>[] = [
    {
      header: (
        <div css={Flex.row}>
          <Heading level={HeadingLevel.h3} fontWeight={FontWeight.Bold}>
            {t("reorder__table_column_title")}
          </Heading>
        </div>
      ),
      key: "title",
      customStyles: {flexGrow: 2},

      content: entity => (
        <Column>
          <Text
            customStyles={[textEllipsis, {cursor: "pointer"}]}
            onClick={() => onEntityClick(entity.id)}
            size={TextSize.Medium}>
            {entity.title}
          </Text>
        </Column>
      )
    },
    isItemsTable
      ? scoringTypeColumn
      : {
          header: (
            <div css={Flex.row}>
              <Heading level={HeadingLevel.h3} fontWeight={FontWeight.Bold}>
                {t("coding_models__detail_items_label")}
              </Heading>
            </div>
          ),
          key: "items",
          content: entity => <Column>{entity.itemsCount}</Column>
        },
    {
      header: (
        <div css={Flex.row}>
          <Heading level={HeadingLevel.h3} fontWeight={FontWeight.Bold}>
            {t("coding_models__detail_maximum_score")}
          </Heading>
        </div>
      ),
      key: "score",
      content: entity => {
        const itemScore = entity.maxScore
        return (
          <Column>
            {itemScore} {itemScore > 0 || maximumScore > 0 ? `(${toPercent(itemScore, maximumScore)}%)` : ""}
          </Column>
        )
      }
    },
    {
      header: (
        <div css={[Flex.row, {justifyContent: "flex-end"}]}>
          <Tooltip title={t("scenario_details__header_button_is_finalized")} inactive={!isReadOnly}>
            <Button
              customStyles={{marginRight: spacingSmall}}
              disabled={disabled || isEmpty(entities) || isReadOnly}
              variant={ButtonVariant.IconOnly}
              onClick={onSortClick}
              icon={IconName.Sort}
            />
          </Tooltip>
          <Tooltip title={t("scenario_details__header_button_is_finalized")} inactive={!isReadOnly}>
            <Button
              onClick={onAddClick}
              variant={ButtonVariant.IconOnly}
              disabled={disabled || isReadOnly}
              icon={IconName.Add}
            />
          </Tooltip>
        </div>
      ),
      key: "delete",
      content: entity => (
        <Columns>
          <Column customStyles={styles.columns.trash}>
            <DeleteOrArchiveEntityButton
              customButtonStyles={deleteEntityButtonStyles.iconOnlyButton}
              iconColor={fontColorLight}
              disabled={isReadOnly}
              onSuccess={() => navigate(deleteEntityConfig.navigateTo)}
              useDeleteHook={() => deleteEntityConfig.deleteEntityHook}
              entityId={entity.id}
            />
          </Column>
        </Columns>
      )
    }
  ]

  return (
    <div className="dimension-table" css={customStyles}>
      <Label label={`${t(labelKey)} (${entities.length})`} />
      <TableContainer<DimensionTableEntity>
        entities={entities}
        entityKey={entity => entity.id}
        showFooter
        customPlaceholder={placeholder}
        customStyles={styles.table}
        placeHolderText={t("coding_models__detail_table_placeholder")}
        customHeaderRowStyles={styles.headerRow}
        columns={columns}
      />
    </div>
  )
}

const styles = {
  headerRow: css({
    padding: spacingSmall
  }),
  columns: {
    trash: css({display: "flex", justifyContent: "flex-end"})
  },
  table: css({
    minHeight: 300 + spacingHuger
  })
}
