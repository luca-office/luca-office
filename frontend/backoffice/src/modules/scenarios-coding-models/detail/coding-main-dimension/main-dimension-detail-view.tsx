import {css} from "@emotion/react"
import * as React from "react"
import {Button, Heading, Text} from "shared/components"
import {HeadingLevel} from "shared/enums"
import {useDeleteCodingDimension, useDeleteCodingItem} from "shared/graphql/hooks"
import {CodingDimension} from "shared/models"
import {Flex, fontColorLight, spacingMedium, spacingSmall, spacingTiny, TextSize} from "shared/styles"
import {useLucaTranslation} from "shared/translations"
import {sortByPosition} from "shared/utils"
import {ResortModal} from "../../../../components"
import {Route} from "../../../../routes"
import {CreateSubDimensionModal} from "../../create"
import {CreateItemModal} from "../../create/create-item/create-item-modal"
import {
  accumulateItemScoreFromDimension,
  countItemsFromDimension,
  toTableItemEntity
} from "../../utils/util-dimension-table"
import {CodingModelCommonDetailViewContainer} from "../common/detail-view/coding-model-common-detail-view-container"
import {DimensionsTable, DimensionTableEntity} from "../common/dimensions-table/dimensions-table"
import {useMainDimensionDetailView} from "./hooks/use-main-dimension-detail-view"

interface Props {
  readonly allCodingDimensions: CodingDimension[]
  readonly codingDimension: CodingDimension
  readonly scenarioId: UUID
  readonly isReadOnly: boolean
}

export const MainDimensionDetailView: React.FC<Props> = ({
  allCodingDimensions,
  codingDimension,
  scenarioId,
  isReadOnly
}) => {
  const {t} = useLucaTranslation()

  const {
    updateCodingDimension,
    setShowCreateSubDimensionsModal,
    setShowCreateItemModal,
    isCreateSubDimensionsModalVisible,
    isCreateItemModalVisible,
    isSortModalVisible,
    isRepositionLoading,
    setIsSortModalVisible,
    navigateToItemDetail,
    navigateToDimensionDetail,
    repositionDimensions,
    repositionItems
  } = useMainDimensionDetailView(codingDimension, scenarioId)

  const maxScore = accumulateItemScoreFromDimension(codingDimension, allCodingDimensions)

  const deleteDimensionHook = useDeleteCodingDimension(codingDimension.codingModelId, scenarioId)
  const deleteItemHook = useDeleteCodingItem(codingDimension.id)

  const subDimensions = allCodingDimensions.filter(dim => dim.parentDimensionId === codingDimension.id)

  const subdimensionsEntities: DimensionTableEntity[] = subDimensions.map(dimension => ({
    id: dimension.id,
    title: dimension.title,
    itemsCount: countItemsFromDimension(dimension, allCodingDimensions),
    maxScore: accumulateItemScoreFromDimension(dimension, allCodingDimensions)
  }))

  const sortedItems: DimensionTableEntity[] = sortByPosition(codingDimension.items.map(toTableItemEntity))

  const dimensionHasItems = codingDimension.items.length > 0

  const emptyTablePlaceholder = (
    <div>
      <Heading customStyles={styles.placeholder} level={HeadingLevel.h3}>
        {t("coding_models__detail_main_dimension_table_placeholder_title")}
      </Heading>
      {!isReadOnly && (
        <>
          <Text size={TextSize.Medium} customStyles={styles.placeholderSub}>
            {t("coding_models__detail_main_dimension_table_placeholder_sub_title")}
          </Text>
          <div css={[Flex.row, styles.buttons]}>
            <Button onClick={() => setShowCreateSubDimensionsModal(true)} customStyles={styles.button}>
              {t("coding_models__detail_sub_dimension_label")}
            </Button>
            <Button onClick={() => setShowCreateItemModal(true)}>{t("coding_models__detail_items_label")}</Button>
          </div>
        </>
      )}
    </div>
  )

  return (
    <>
      <CodingModelCommonDetailViewContainer
        isReadOnly={isReadOnly}
        headerTitleKey="coding_models__detail_main_dimension_label_singular"
        editDescriptionDialogTitleKey="coding_models__detail_main_dimension_edit_description_modal"
        description={codingDimension.description}
        descriptionPlaceholderKey="coding_models__create_main_dimension_description_placeholder"
        title={codingDimension.title}
        headerDeleteButtonConfig={{
          deleteEntityHook: deleteDimensionHook,
          entityId: codingDimension.id,
          navigateTo: {
            route: Route.ScenarioCodingModelDetail,
            payload: {codingModelId: codingDimension.codingModelId, scenarioId}
          }
        }}
        renderCustomContent={() => (
          <DimensionsTable
            isReadOnly={isReadOnly}
            placeholder={emptyTablePlaceholder}
            onEntityClick={entityId =>
              dimensionHasItems ? navigateToItemDetail(entityId) : navigateToDimensionDetail(entityId)
            }
            onAddClick={() =>
              dimensionHasItems ? setShowCreateItemModal(true) : setShowCreateSubDimensionsModal(true)
            }
            onSortClick={() => setIsSortModalVisible(true)}
            isItemsTable={dimensionHasItems}
            customStyles={styles.dimensionsTable}
            maximumScore={maxScore}
            deleteEntityConfig={{
              deleteEntityHook: dimensionHasItems ? deleteItemHook : deleteDimensionHook,
              navigateTo: {
                route: Route.ScenarioCodingDimensionDetail,
                payload: {scenarioId, codingModelId: codingDimension.codingModelId, dimensionId: codingDimension.id}
              }
            }}
            labelKey={
              dimensionHasItems ? "coding_models__detail_items_label" : "coding_models__detail_sub_dimension_label"
            }
            entities={dimensionHasItems ? sortedItems : subdimensionsEntities}
          />
        )}
        maxScore={maxScore}
        handleUpdate={updateCodingDimension}
      />
      {isCreateSubDimensionsModalVisible && (
        <CreateSubDimensionModal
          scenarioId={scenarioId}
          onConfirm={() => setShowCreateSubDimensionsModal(false)}
          onDismiss={() => setShowCreateSubDimensionsModal(false)}
          codingModelId={codingDimension.codingModelId}
          parentDimension={codingDimension}
        />
      )}
      {isCreateItemModalVisible && (
        <CreateItemModal
          scenarioId={scenarioId}
          codingModelId={codingDimension.codingModelId}
          onConfirm={() => setShowCreateItemModal(false)}
          onDismiss={() => setShowCreateItemModal(false)}
          parentDimension={codingDimension}
        />
      )}
      {isSortModalVisible && (
        <ResortModal<DimensionTableEntity>
          entities={dimensionHasItems ? sortedItems : subdimensionsEntities}
          disabled={isRepositionLoading}
          onConfirm={dimensionHasItems ? repositionItems : repositionDimensions}
          onDismiss={() => setIsSortModalVisible(false)}
          descriptionTextKey={
            dimensionHasItems
              ? "coding_models__detail_item_sort_modal_description"
              : "coding_models__detail_sub_dimension_sort_modal_description"
          }
          titleKey={
            dimensionHasItems
              ? "coding_models__detail_item_sort_modal_title"
              : "coding_models__detail_sub_dimension_sort_modal_title"
          }
          tableLabel={
            dimensionHasItems
              ? `${t("coding_models__detail_items_label")} (${sortedItems.length})`
              : `${t("coding_models__detail_sub_dimension_label")} (${subdimensionsEntities.length})`
          }
        />
      )}
    </>
  )
}

const styles = {
  placeholder: css({
    marginBottom: spacingTiny
  }),
  placeholderSub: css({
    marginBottom: spacingMedium,
    color: fontColorLight
  }),
  buttons: css({
    justifyContent: "center"
  }),
  button: css({
    marginRight: spacingSmall
  }),
  dimensionsTable: {
    marginTop: spacingMedium
  }
}
