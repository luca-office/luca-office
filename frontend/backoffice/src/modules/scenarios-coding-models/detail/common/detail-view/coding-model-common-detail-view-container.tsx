import * as React from "react"
import {DeleteEntityHook, NavigationConfig} from "shared/models"
import {LucaI18nLangKey, LucaTFunction, useLucaTranslation} from "shared/translations"
import {useNavigate} from "../../../../../hooks/use-navigate"
import {Route} from "../../../../../routes"
import {CommonCodingModelUpdate} from "../../overview/coding-model-detail-overview-container"
import {CodingModelCommonDetailView} from "./coding-model-common-detail-view"

export interface CodingModelCommonDetailViewComponentProps extends CodingModelCommonDetailViewContainerProps {
  readonly navigate: (navigationConfig: NavigationConfig<Route>) => void
  readonly t: LucaTFunction
}

export interface CodingModelCommonDetailViewContainerProps {
  readonly description: string
  readonly descriptionPlaceholderKey: LucaI18nLangKey
  readonly isReadOnly: boolean
  readonly handleUpdate: (update: CommonCodingModelUpdate) => void
  readonly headerTitleKey: LucaI18nLangKey
  readonly editDescriptionDialogTitleKey: LucaI18nLangKey
  readonly maxScore: number
  readonly headerDeleteButtonConfig?: {
    readonly entityId: UUID
    readonly deleteEntityHook: DeleteEntityHook
    readonly navigateTo: NavigationConfig<Route>
  }
  readonly renderCustomContent?: () => JSX.Element
  readonly title: string
}

export const CodingModelCommonDetailViewContainer = (props: CodingModelCommonDetailViewContainerProps) => {
  const {navigate} = useNavigate()

  const {t} = useLucaTranslation()
  return <CodingModelCommonDetailView t={t} navigate={navigate} {...props} />
}
