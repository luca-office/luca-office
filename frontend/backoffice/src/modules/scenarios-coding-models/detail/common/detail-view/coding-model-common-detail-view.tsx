import {css} from "@emotion/react"
import * as React from "react"
import {
  Card,
  CardHeader,
  deleteEntityButtonStyles,
  DeleteOrArchiveEntityButton,
  Heading,
  Label
} from "shared/components"
import {HeadingLevel, IconName} from "shared/enums"
import {CodingModelCreation} from "shared/graphql/generated/globalTypes"
import {fontColorLight, spacingLarge, spacingMedium, spacingSmall} from "shared/styles"
import {
  InlineEditableHeaderContainer,
  InlineEditableTextareaContainer,
  OverlayEditField,
  OverlayEditFieldType
} from "../../../../../components"
import {CodingModelDetailViewFooter} from "../coding-model-detail-view-footer"
import {CodingModelCommonDetailViewComponentProps} from "./coding-model-common-detail-view-container"

export const CodingModelCommonDetailView: React.FC<CodingModelCommonDetailViewComponentProps> = ({
  description,
  handleUpdate,
  headerTitleKey,
  descriptionPlaceholderKey,
  maxScore,
  renderCustomContent,
  editDescriptionDialogTitleKey,
  isReadOnly,
  navigate,
  t,
  headerDeleteButtonConfig,
  title
}) => {
  return (
    <Card customStyles={{overflowY: "auto"}}>
      <CardHeader customStyles={styles.header} hasGreyBackground hasShadow>
        <Heading level={HeadingLevel.h3}>{t(headerTitleKey)}</Heading>
        {headerDeleteButtonConfig && (
          <DeleteOrArchiveEntityButton
            disabled={isReadOnly}
            customStyles={styles.deleteEntityButton}
            iconColor={fontColorLight}
            entityId={headerDeleteButtonConfig.entityId}
            customButtonStyles={deleteEntityButtonStyles.iconOnlyButton}
            onSuccess={() =>
              navigate({
                route: headerDeleteButtonConfig.navigateTo.route,
                payload: headerDeleteButtonConfig.navigateTo.payload
              })
            }
            useDeleteHook={() => headerDeleteButtonConfig.deleteEntityHook}
          />
        )}
      </CardHeader>
      <div css={styles.content}>
        <div css={styles.title}>
          <Label label={t("title")} icon={isReadOnly ? IconName.LockClosed : IconName.EditPencil} />
          <InlineEditableHeaderContainer
            onConfirm={title => handleUpdate({title})}
            text={title}
            disabled={isReadOnly}
            customStyles={styles.inlineField}
          />
        </div>
        <Label label={t("description")} icon={isReadOnly ? IconName.LockClosed : IconName.EditPencil} />
        <OverlayEditField<CodingModelCreation>
          formFields={[
            {
              updateId: "description",
              value: description,
              labelKey: "description",
              type: OverlayEditFieldType.TEXTAREA
            }
          ]}
          fieldLabelKey={"description"}
          customStyles={styles.descriptionField}
          dialogTitleKey={editDescriptionDialogTitleKey}
          onUpdate={value => handleUpdate({description: value.description})}
          updateLoading={false}
          disabled={isReadOnly}
          renderValue={() => (
            <InlineEditableTextareaContainer
              placeholder={t(descriptionPlaceholderKey)}
              text={description}
              disabled={isReadOnly}
              readOnly={true}
            />
          )}
          displayPlain={true}
        />

        {renderCustomContent?.()}
      </div>
      <CodingModelDetailViewFooter totalScore={maxScore} />
    </Card>
  )
}

const styles = {
  header: css({
    justifyContent: "space-between"
  }),
  content: css({
    padding: spacingMedium,
    height: "100%",
    overflowY: "auto"
  }),
  title: css({
    marginBottom: spacingLarge
  }),
  deleteEntityButton: css({
    cursor: "pointer"
  }),
  inlineField: css({
    marginLeft: -spacingSmall
  }),
  descriptionField: css({
    minHeight: 58,
    marginLeft: -spacingSmall
  })
}
