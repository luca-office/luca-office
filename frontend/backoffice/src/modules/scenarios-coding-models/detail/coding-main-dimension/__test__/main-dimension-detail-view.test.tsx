// importing from direct file because of issues of babel loader and spyOn
import {MockedProvider} from "@apollo/client/testing"
import {mount} from "enzyme"
import * as React from "react"
import {create} from "react-test-renderer"
import {Button, Heading} from "shared/components"
import {codingDimensionsMock} from "shared/graphql/__mocks__"
import {CodingModelCommonDetailView} from "../../common/detail-view/coding-model-common-detail-view"
import {DimensionsTable} from "../../common/dimensions-table/dimensions-table"
import {MainDimensionDetailView} from "../main-dimension-detail-view"

describe("main-dimension-detail-view", () => {
  it("renders correctly", async () => {
    const component = create(
      <MockedProvider>
        <MainDimensionDetailView
          isReadOnly={false}
          scenarioId="sdfsdf-sdflsk"
          codingDimension={codingDimensionsMock[0]}
          allCodingDimensions={codingDimensionsMock}
        />
      </MockedProvider>
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
  it("has correct structure", async () => {
    const component = mount(
      <MockedProvider>
        <MainDimensionDetailView
          isReadOnly={false}
          scenarioId="sdfsdf-sdflsk"
          codingDimension={codingDimensionsMock[0]}
          allCodingDimensions={codingDimensionsMock}
        />
      </MockedProvider>
    )

    expect(component.find(CodingModelCommonDetailView)).toHaveLength(1)
    expect(component.find(Heading)).toHaveLength(4)
    expect(component.find(DimensionsTable)).toHaveLength(1)
    expect(component.find(DimensionsTable).prop("isItemsTable")).toBe(true)
    expect(component.find(Button)).toHaveLength(4)
  })

  it("has correct structure without items", async () => {
    const component = mount(
      <MockedProvider>
        <MainDimensionDetailView
          isReadOnly={false}
          scenarioId="sdfsdf-sdflsk"
          codingDimension={codingDimensionsMock[1]}
          allCodingDimensions={codingDimensionsMock}
        />
      </MockedProvider>
    )

    expect(component.find(DimensionsTable).prop("isItemsTable")).toBe(false)
  })
})
