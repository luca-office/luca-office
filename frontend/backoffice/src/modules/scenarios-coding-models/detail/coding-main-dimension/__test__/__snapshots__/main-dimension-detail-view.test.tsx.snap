// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`main-dimension-detail-view renders correctly 1`] = `
<div
  className="css-1xygdyb-Card"
>
  <div
    className="card-header css-szn5yk-CardHeader"
  >
    <div
      className="css-1f6p2vj-Heading"
    >
      coding_models__detail_main_dimension_label_singular
    </div>
    <div
      className="css-dw2z8l-OrlyButton"
    >
      <button
        className="css-qsdvj6-Button"
        disabled={false}
        onClick={[Function]}
      >
        <div
          className="css-95j2hm-Button"
        >
          <div
            className="css-p7iffv-Icon"
            dangerouslySetInnerHTML={
              {
                "__html": {},
              }
            }
          />
        </div>
      </button>
    </div>
  </div>
  <div
    className="css-4toq1p-CodingModelCommonDetailView"
  >
    <div
      className="css-12ryqhc-CodingModelCommonDetailView"
    >
      <div
        className="css-lz4b3k-Label"
      >
        title
        <div
          className="css-nhflpf-Icon"
          dangerouslySetInnerHTML={
            {
              "__html": {},
            }
          }
        />
      </div>
      <div
        className="css-1yd7xcl-InlineEditableHeader"
      >
        <div>
          <form
            className="css-1c5ej61-InlineEditableHeader"
            onBlur={[Function]}
            onSubmit={[Function]}
          >
            <div
              className="css-pl4s5v-InlineEditableHeader"
              onClick={[Function]}
            >
              <div
                className="css-1iwo50k"
                onMouseEnter={[Function]}
                onMouseLeave={[Function]}
              >
                <div
                  className="css-nrul0i"
                >
                  <input
                    className="css-q6vozw"
                    defaultValue="Argumentation (inhaltlich)"
                    disabled={false}
                    onBlur={[Function]}
                    onChange={[Function]}
                    onFocus={[Function]}
                    readOnly={true}
                    type="text"
                  />
                </div>
              </div>
              <div
                className="hover-icon css-16vb3lw-Icon"
                dangerouslySetInnerHTML={
                  {
                    "__html": {},
                  }
                }
              />
            </div>
          </form>
        </div>
      </div>
    </div>
    <div
      className="css-lz4b3k-Label"
    >
      description
      <div
        className="css-nhflpf-Icon"
        dangerouslySetInnerHTML={
          {
            "__html": {},
          }
        }
      />
    </div>
    <div
      className="overlay-edit-field css-0"
    >
      <div
        className="css-agvbnc-ReadonlyActionField"
        data-testid="readonly-action-field"
        onClick={[Function]}
      >
        <div
          className="css-1bkess4-ReadonlyActionField"
        />
        <div
          className="content-wrapper css-376q3o-ReadonlyActionField"
        >
          <div
            className="content css-ci7wec-ReadonlyActionField"
          >
            <div
              className="css-1h2eymq-InlineEditableTextarea"
              onClick={[Function]}
            >
              <div
                className="textarea css-1n5bn12-InlineEditableTextarea"
                contentEditable={false}
                suppressContentEditableWarning={true}
              >
                dsfsdf
              </div>
              <div
                className="hover-icon css-1pndqgp-Icon"
                dangerouslySetInnerHTML={
                  {
                    "__html": {},
                  }
                }
              />
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="dimension-table css-wt5pfw-DimensionsTable"
    >
      <div
        className="css-lz4b3k-Label"
      >
        coding_models__detail_items_label (1)
      </div>
      <div
        className="css-uou2au-Table"
      >
        <div
          className="css-v9uh6m-Table"
        >
          <div
            className="css-kc008c-Table"
          >
            <div
              className="css-1r5hlla-DimensionsTable"
            >
              <div
                className="css-8050ac-Heading"
              >
                reorder__table_column_title
              </div>
            </div>
          </div>
          <div
            className="css-fuywjr-Table"
          >
            <div
              className="css-1r5hlla-DimensionsTable"
            >
              <div
                className="css-8050ac-Heading"
              >
                coding_models__detail_items_scoring_type
              </div>
            </div>
          </div>
          <div
            className="css-fuywjr-Table"
          >
            <div
              className="css-1r5hlla-DimensionsTable"
            >
              <div
                className="css-8050ac-Heading"
              >
                coding_models__detail_maximum_score
              </div>
            </div>
          </div>
          <div
            className="css-fuywjr-Table"
          >
            <div
              className="css-1ea805l-DimensionsTable"
            >
              <button
                className="css-1swaics-Button"
                disabled={false}
                onClick={[Function]}
              >
                <div
                  className="css-95j2hm-Button"
                >
                  <div
                    className="css-mj8vo1-Icon"
                    dangerouslySetInnerHTML={
                      {
                        "__html": {},
                      }
                    }
                  />
                </div>
              </button>
              <button
                className="css-1cywa3f-Button"
                disabled={false}
                onClick={[Function]}
              >
                <div
                  className="css-95j2hm-Button"
                >
                  <div
                    className="css-mj8vo1-Icon"
                    dangerouslySetInnerHTML={
                      {
                        "__html": {},
                      }
                    }
                  />
                </div>
              </button>
            </div>
          </div>
        </div>
        <div
          className="css-tqgzls-Table"
        >
          <div
            className="css-1ihrgr1-Table"
          >
            <div
              className="entity-row css-hij9lk-Table"
              onClick={[Function]}
            >
              <div
                className="css-kc008c-Table"
              >
                <div
                  className="css-1oiyj7r-Column"
                >
                  <div
                    className="css-c44d5y-Text"
                    onClick={[Function]}
                  >
                    Item 1
                  </div>
                </div>
              </div>
              <div
                className="css-fuywjr-Table"
              >
                <div
                  className="css-1oiyj7r-Column"
                >
                  coding_models__detail_items_scoring_type_analytic
                </div>
              </div>
              <div
                className="css-fuywjr-Table"
              >
                <div
                  className="css-1oiyj7r-Column"
                >
                  11
                   
                  (100%)
                </div>
              </div>
              <div
                className="css-fuywjr-Table"
              >
                <div
                  className="css-1cky3ub-Columns"
                >
                  <div
                    className="css-hqw4y8-Column"
                  >
                    <div
                      className="css-0"
                    >
                      <button
                        className="css-qsdvj6-Button"
                        disabled={false}
                        onClick={[Function]}
                      >
                        <div
                          className="css-95j2hm-Button"
                        >
                          <div
                            className="css-p7iffv-Icon"
                            dangerouslySetInnerHTML={
                              {
                                "__html": {},
                              }
                            }
                          />
                        </div>
                      </button>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        className="card-footer css-jb8qoz-CardFooter"
      />
    </div>
  </div>
  <div
    className="card-footer css-gmu9kx-CardFooter"
  >
    <div
      className="css-15tad5r-CodingModelDetailViewFooter"
    >
      <div
        className="css-17zbm7y-Text"
      >
        coding_models__detail_maximum_score_long
      </div>
      <div
        className="css-1m37bd5-CodingModelDetailViewFooter"
      >
        <div
          className="css-cn392y-Text"
        >
          11
           
          coding_models__detail_score
        </div>
      </div>
    </div>
  </div>
</div>
`;
